src/resource.jl

```diff
@@ -270,42 +270,58 @@ function forget_failures()
     end
 end
 
-function tarball_git_hash(tarball::String)
-    local tree_hash
-    mktempdir() do tmp_dir
-        run(`tar -C $tmp_dir -zxf $tarball`)
-        tree_hash = bytes2hex(Pkg.GitTools.tree_hash(tmp_dir))
-        chmod(tmp_dir, 0o777, recursive=true)
-    end
-    return tree_hash
-end
-
 function download(server::String, resource::String)
     @info "downloading resource" server=server resource=resource
     hash = let m = match(hash_part_re, resource)
         m !== nothing ? m.captures[1] : nothing
     end
 
-    write_atomic_lru(resource) do temp_file, io
-        response = HTTP.get(
+    write_atomic_lru(resource) do temp_file, file_io
+        buffio = Base.BufferStream()
+        tar_check_io = Base.BufferStream()
+        tar_extract_task = @async begin
+            # Only do this work if hash !=== nothing
+            if hash === nothing
+                return nothing
+            end
+            @info("tar_extract_task", resource, hash)
+            Tar.tree_hash(TranscodingStream(GzipDecompressor(), tar_check_io))
```

Comment:
I wonder if we should just use an external `gzip` process for this because of the CodecGzip memory issues. Either that or I should spend some time fixing the issues, which didn't seem super trivial and then needs to be repeated for all the CodecX packages 🙁. I have a suspicion that using an external process may be both safer (in terms of memory) and faster. If we moreover were to start the `gzip` process in a sandbox with only the ability to receive data on stdin and produce results on stdout (and error messages on stderr), then it would be safer in the security sense as well, which also would not be a bad thing.

Reply:
For now, I will stream it through a `gzip` external command, but I agree that we should fix the `CodecX` packages.  Perhaps we should open an issue on TranscodingStreams and notify Kenta about our issues and a proposed solution.

Reply:
Based on past issues I've opened, I don't really expect him to work on this. I think it's on me.